packages/api/src/controllers/asset.ts

```diff
@@ -106,12 +108,49 @@ function cleanAssetTracks(asset: WithID<Asset>) {
       };
 }
 
+function parseUrlToDStorageUrl(
```

Comment:
Yeah i wanted something like that but couldn't find one for arweave. Since we already gate this transformation through a trusted gateway list though, we can be pretty sure that if the CID is invalid the import is going to fail and the user not gate a fake association with a CID or tx ID. So that should be enough for now.